taichi/ui/common/window_base.h

```diff
@@ -41,6 +42,8 @@ class WindowBase {
 
   virtual void write_image(const std::string &filename) = 0;
 
+  virtual void copy_depth_buffer_to_ndarray(taichi::lang::Ndarray *) = 0;
```

Comment:
I feel like it's possible to use `const taichi::lang::Ndarray& ndarray)` , could you give that a try?